portainer/app/build.py

```diff
@@ -63,6 +63,9 @@ def main(args):
     if args.framework_id:
         framework.id.value = args.framework_id
 
+    if args.framework_role:
+        framework.role.value = args.framework_role
```

Comment:
The `role` field in the `FrameworkInfo` is a string, not another structure, so you can just assign to `framework.role`.

See https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L155